integration/query_projection_test.go

```diff
@@ -252,3 +252,35 @@ func TestQueryProjectionErrors(t *testing.T) {
 		})
 	}
 }
+
+func TestQueryProjectionSuccess(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup.Setup(t, shareddata.Scalars, shareddata.Composites)
+	for name, tc := range map[string]struct { //nolint:vet // used for testing only
+		filter     bson.D // required
+		projection any    // required
+
+		skip       string              // optional, skip test with a specified reason
```

Comment:
Please don't forget to run `task all`, there is format warning. For quicker check for just formatting and linting, you could also run `task` fmt` and `task lint`. They are run by `task all` too :pray: